packages/babel-parser/src/parser/expression.js

```diff
@@ -1138,6 +1138,9 @@ export default class ExpressionParser extends LValParser {
           this.registerTopicReference();
           return this.finishNode(node, "PipelinePrimaryTopicReference");
         }
+
+        this.expectPlugin("privateIn");
+        return this.parseMaybePrivateName(true);
```

Comment:
Can we add
```js
if (!this.match(tt._in)) this.raise
```
?

Reply:
And we should also check if the private name is parenthesized.
```js
(#p) in C
```
seems invalid.

Reply:
Done and done.